File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    student1 = [1, 2, 3, 4, 5]
    student2 = [2, 1, 2, 3, 2, 4, 2, 5]
    student3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    score = [0, 0, 0]
    
    for idx, ans in enumerate(answers):
        if ans == student1[idx % len(student1)]:
            score[0] += 1
        if ans == student2[idx % len(student2)]:
            score[1] += 1
        if ans == student3[idx % len(student3)]:
            score[2] += 1
            
    for idx, sc in enumerate(score):
        if sc == max(score):
            answer.append(idx+1)
            
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    let p1 = [1, 2, 3, 4, 5]
    let p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    let p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    var cnt = [0,0,0]
    
    for (idx,ans) in answers.enumerated() {
        if p1[idx%p1.count] == ans { cnt[0] += 1 }
        if p2[idx%p2.count] == ans { cnt[1] += 1 }
        if p3[idx%p3.count] == ans { cnt[2] += 1 }
    }
    
    let max = cnt.max()
    var answer = [Int]()
    for (idx,c) in cnt.enumerated() {
        if c == max { answer.append(idx+1)}
    }
    
    return answer.sorted()
}